Sources/Moya/Task+Internal.swift

```diff
@@ -0,0 +1,86 @@
+import Foundation
+import Moya
+
+extension Task {
+    typealias TaskParameters = (Encodable, ParameterEncoder)
+
+    func allParameters() throws -> [TaskParameters] {
+
+        var providers: [TaskParametersProvider?] = [bodyParams, urlParams]
```

Comment:
Why not? 

```swift 
var providers: [TaskParametersProvider?] = [bodyParams, urlParams, customParams]
```

You are filtering the nil values anyway.

Reply:
customParams is `[TaskParametersProvider]` so I can't mix it with `TaskParametersProvider` values.

Reply:
Missed that! How about this?

```swift 
var providers: [TaskParametersProvider?] = customParams.append(contentsOf: [bodyParams, urlParams])
```

Reply:
`append` mutates the receiver but doesn't return anything.

This would work but I find it less readable than the original:
```
var providers: [TaskParametersProvider?] = customParams ?? []
providers.append(contentsOf: ([bodyParams, urlParams] as [TaskParametersProvider?]))
```

Reply:
It would have been nice if Swift supported something like `appending(contentsOf:)` 😄.
But, never mind then.